streaming/base/dataset.py

```diff
@@ -1062,12 +1062,15 @@ def get_item(self, sample_id: int, retry: int = 7) -> Any:
 
                 # On success, break out.
                 break
-            except:
-                # Fallback: ensure the shard is downloaded, then try to access the sample again.
+            except FileNotFoundError:
+                # Fallback: shard file is missing (generates `FileNotFoundError` exception),
+                # ensure the shard file is downloaded, then try to access the sample again.
                 # Loops because it may become evicted in the meantime.
                 self.download_shard(shard_id)
         else:
-            raise RuntimeError('StreamingDataset is thrashing. Raise cache_limit.')
+            raise RuntimeError(f'StreamingDataset is thrashing. Either an issue with ' +
```

Comment:
can we change this to two errors depending on cache_limit:

if self.cache_limit:
    say "StreamingDataset repeatedly failed to download a shard. This may be due to thrashing caused by `cache_limit` being set too low"
else:
    just say "StreamingDataset repeatedly failed to download a shard."

Reply:
Updated in https://github.com/mosaicml/streaming/pull/298/commits/ffb9eb9140df12f9f855fc02b82bf98c60ba776b.